src/menu.h

```diff
@@ -0,0 +1,503 @@
+        #endif
+    } 
+    else {
+        if (xQueueReceive(button_events, &ev, 1/portTICK_PERIOD_MS)) {
+            if (ev.event == BUTTON_UP) {
+                #if ROTARY_MENU_DEBUG == 1
+                    debugPrintf("Processing Click");
+                #endif
+                LCDML.BT_enter();
+            } else if (ev.event == BUTTON_HELD) {
+                long abortTime = millis();
+                if (abortTime - lastMenuAbort > menuAbortInterval) {
+                    #if ROTARY_MENU_DEBUG == 1
+                        debugPrintf("Processing button held, aborting.");
+                    #endif
+                    LCDML.FUNC_goBackToMenu(1);
+                    lastMenuAbort = abortTime;
+                }
+            }
+        }
+    }
+
+    last = pos;
+}
+
+void clearMenu() {
+}
+
+void setupMenu() {
+    LCDML_setup(_LCDML_DISP_cnt);
+}
+
+void displayMenu() {
+    #if ROTARY_MENU_DEBUG // output menu to serial 
+    // init vars
+    //uint8_t n_max = (LCDML.MENU_getChilds() >= _LCDML_DISP_rows) ? _LCDML_DISP_rows : (LCDML.MENU_getChilds());
+
+    // update content
+    // ***************
+    if (LCDML.DISP_checkMenuUpdate() || LCDML.DISP_checkMenuCursorUpdate() ) {
+        // clear menu
+        // ***************
+        LCDML.DISP_clear();
+
+        debugPrintln(F("==========================================="));
+        debugPrintln(F("================  Menu ===================="));
+        debugPrintln(F("==========================================="));
+
+        // declaration of some variables
+        // ***************
+        // content variable
+        char content_text[_LCDML_DISP_cols];  // save the content text of every menu element
+        // menu element object
+        LCDMenuLib2_menu *tmp;
+        // some limit values
+        uint8_t i = LCDML.MENU_getScroll();
+        uint8_t maxi = (_LCDML_DISP_rows) + i;
+        uint8_t n = 0;
+
+        // check if this element has children
+        if ((tmp = LCDML.MENU_getDisplayedObj()) != NULL)
+        {
+
+        // loop to display lines
+        do
+        {
+            // check if a menu element has a condition and if the condition be true
+            if (tmp->checkCondition())
+            {
+            // display cursor
+            if (n == LCDML.MENU_getCursorPos())
+            {
+                debugPrint(F("(x) "));
+            }
+            else
+            {
+                debugPrint(F("( ) "));
+            }
+
+            // check the type off a menu element
+            if(tmp->checkType_menu() == true)
+            {
+                // display normal content
+                LCDML_getContent(content_text, tmp->getID());
+                debugPrint(content_text);
+            }
+            else
+            {
+                if(tmp->checkType_dynParam()) {
+                tmp->callback(n);
+                }
+            }
+
+            debugPrintln("");
+
+            // increment some values
```

Comment:
This comment contains no information that isn't perfectly obvious from the 2 lines of code below :)